core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -450,10 +450,9 @@ private[deploy] class Worker(
         }
       }(cleanupThreadExecutor)
 
-      cleanupFuture.onFailure {
-        case e: Throwable =>
-          logError("App dir cleanup failed: " + e.getMessage, e)
-      }(cleanupThreadExecutor)
+      cleanupFuture.failed.foreach(e =>
```

Comment:
onFailure, onSuccess are deprecated in 2.12. This should be equivalent in 2.11+2.12